burn-import/src/pytorch/adapter.rs

```diff
@@ -0,0 +1,107 @@
+use burn::{
+    module::Param,
+    record::{PrecisionSettings, Record},
+    tensor::{backend::Backend, Tensor},
+};
+
+use burn::record::serde::{
+    adapter::{BurnModuleAdapter, DefaultAdapter},
+    data::NestedValue,
+    ser::Serializer,
+};
+
+use burn::backend::NdArray;
+
+use serde::Serialize;
+
+/// A dummy backend used to transform tensors.
+type DummyBackend = NdArray<f32>;
+
+/// A PyTorch adapter for the Burn module used during deserialization.
+///
+/// Not all Burn module correspond to a PyTorch module. Therefore,
+/// we need to adapt the Burn module to a PyTorch module. We implement
+/// only those that differ.
+pub struct PyTorchAdapter<PS: PrecisionSettings> {
+    _precision_settings: std::marker::PhantomData<PS>,
+}
+
+impl<PS: PrecisionSettings> BurnModuleAdapter for PyTorchAdapter<PS> {
+    fn adapt_linear(data: NestedValue) -> NestedValue {
+        // Get the current module in the form of map.
+        let mut map = data.get_map().expect("Failed to get map from NestedValue");
+
+        // Get/remove the weight parameter.
+        let weight = map
+            .remove("weight")
+            .expect("Failed to find 'weight' key in map");
+
+        // Convert the weight parameter to a tensor.
+        let weight: Param<Tensor<DummyBackend, 2>> = weight
```

Comment:
I don't think we should use a DummyBackend, maybe add the backend as a generic argument to `BurnModuleAdapter` ?

Reply:
OK